indexer/tasks/src/multiera/multiera_address.rs

```diff
@@ -131,15 +132,15 @@ fn queue_certificate(
     queued_address_credential: &mut BTreeSet<QueuedAddressCredentialRelation>,
     queued_address: &mut BTreeMap<Vec<u8>, i64>,
     tx_id: i64,
-    cert: &MultiEraCert,
+    cert: &MultiEraCertificate,
 ) {
     // TODO: what's the policy for handling options? At the moment of writing, all certificates
     // are "alonzo-compatible", but that might change in a future HF. Should Carp skip data that
     // it doesn't understand or instead panic? For now, opting to panic as it seems to be what's
     // used for other options.
-    match cert.as_alonzo().unwrap() {
-        Certificate::StakeDelegation(credential, pool) => {
-            let credential = credential.encode_fragment().unwrap();
+    match cert {
+        MultiEraCertificate::StakeDelegation(delegation) => {
+            let credential = delegation.stake_credential.to_canonical_cbor_bytes();
```

Comment:
note that `to_cannonical` isn't the same representation of what is onchain. I don't think that's necessarily a bad thing for fields that aren't critical (block and transactions), but if we decide to use canonical instead of the real encoding onchain we need to specify that somewhere

This comment applies not just here, but other places as well

Reply:
we can switch to `.to_cbor_bytes()` for safety

Reply:
Yeah I would rather we use the same encoding as what is onchain whenever possible instead of converting it to cannonical cbor

Reply:
ok fixed